mlbench_core/optim/pytorch/optim.py

```diff
@@ -499,12 +499,290 @@ def step(self, closure=None):
         return loss
 
 
+class PowerSGD(Optimizer):
+    def __init__(
+        self,
+        model=None,
+        lr=0.1,
+        weight_decay=0,
+        momentum=0,
+        dampening=0,
+        nesterov=False,
+        reuse_query=False,
+        rank=1,
+    ):
```

Comment:
looks nice! in other places where rank is used for a different meaning, comment to clarify. and here clarify that its matrix approximation rank

Reply:
In the other classes, the meaning of `rank` is explained inside a docstring. I will do that also for this class.

Reply:
Please always add google style docstrings to all non-private methods at the least and ideally to all methods. http://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Reply:
documentation comment: could someone make a separate issue to more clearly document node id/rank parameters as "unique id/rank of this node" in all other mlbench places. this would avoid confusion with low-rank in the linear algebra sense here